WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductInventoryViewModel.kt

```diff
@@ -38,7 +38,12 @@ class ProductInventoryViewModel @AssistedInject constructor(
             inventoryData = navArgs.inventoryData,
             isIndividualSaleSwitchVisible = isProduct,
             isStockManagementVisible = !isProduct || navArgs.productType != EXTERNAL && navArgs.productType != GROUPED,
-            isStockStatusVisible = !isProduct || navArgs.productType != VARIABLE
+            isStockStatusVisible = !isProduct || navArgs.productType != VARIABLE,
+
+            // Stock quantity field is only editable if the value is whole decimal (e.g: 10.0).
+            // Otherwise it is set to read-only, because the API doesn't support updating amount with non-zero
+            // fractional yet.
+            isStockQuantityEditable = navArgs.inventoryData.stockQuantity?.rem(1)?.equals(0.0)
```

Comment:
Maybe we could extract this to method, I think it might be easier to read. I'm thinking of something like

```suggestion
            isStockQuantityEditable = navArgs.inventoryData.stockQuantity?.isInteger()
```

and somewhere:
```kotlin
private fun Double?.isInteger() = this?.rem(1) == 0.0
```

I've also changed `equals` to `==` as it's the same thing effectively in Kotlin.

Reply:
Nice suggestion ❤️ ! Fixed in 51e6aa3